index.js

```diff
@@ -23,77 +20,57 @@ async function run() {
         owner: owner,
         repo: repo,
       })
+
       if (latestRelease) {
         baseRef = latestRelease.data.tag_name
       } else {
-        core.setFailed(
-          `There are no releases on ${owner}/${repo}. Tags are not releases.`,
-        )
+        core.setFailed(`There are no releases on ${owner}/${repo}. Tags are not releases.`)
       }
     }
 
-    console.log(`head-ref: ${headRef}`)
-    console.log(`base-ref: ${baseRef}`)
+    if (!!headRef && !!baseRef && regexp.test(headRef) && regexp.test(baseRef)) {
+      var resp
 
-    if (
-      !!headRef &&
-      !!baseRef &&
-      regexp.test(headRef) &&
-      regexp.test(baseRef)
-    ) {
-      getChangelog(headRef, baseRef, owner + '/' + repo)
-    } else {
-      core.setFailed(
-        'Branch names must contain only numbers, strings, underscores, periods, and dashes.',
-      )
-    }
-  } catch (error) {
-    core.setFailed(error.message)
-  }
-}
-
-async function getChangelog(headRef, baseRef, repoName) {
-  try {
-    let output = ''
-    let err = ''
-
-    // These are option configurations for the @actions/exec lib`
-    const options = {}
-    options.listeners = {
-      stdout: (data) => {
-        output += data.toString()
-      },
-      stderr: (data) => {
-        err += data.toString()
-      },
-    }
-    options.cwd = './'
-
-    await exec.exec(
-      `${src}/changelog.sh`,
-      [headRef, baseRef, repoName],
-      options,
-    )
+      try {
+        resp = await generateReleaseNotes(owner, repo, baseRef, headRef)
```

Comment:
Could we directly call octokit here instead of our wrapper? The API is basically the same.

https://octokit.github.io/rest.js/v18#repos-generate-release-notes